src/libproc_macro/lib.rs

```diff
@@ -609,110 +655,164 @@ impl Group {
     }
 }
 
+/// Prints the group as a string that should be losslessly convertible back
+/// into the same group (modulo spans), except for possibly `TokenTree::Group`s
+/// with `Delimiter::None` delimiters.
 #[unstable(feature = "proc_macro", issue = "38356")]
 impl fmt::Display for Group {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         TokenStream::from(TokenTree::from(self.clone())).fmt(f)
     }
 }
 
-/// An `Op` is an operator like `+` or `-`, and only represents one character.
+/// An `Punct` is an single punctuation character like `+`, `-` or `#`.
 ///
-/// Operators like `+=` are represented as two instance of `Op` with different
+/// Multicharacter operators like `+=` are represented as two instances of `Punct` with different
 /// forms of `Spacing` returned.
+///
+/// REVIEW ATTENTION: `Copy` impl on a struct with private fields.
+/// REVIEW Do we want to guarantee `Punct` to be `Copy`?
```

Comment:
I think this is less critical than `Span` to be copy, and wouldn't mind going either way here. Making this not-`Copy` is probably the most conservative thing to do

Reply:
I agree, let's go with no Copy.